src/containers/BooksForm.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { createBook } from '../actions';
+
+const styles = {
+  input: {
+    marginRight: "10px",
+    height: "45px",
+    width: "60%",
+    fontSize: "18px"
+  },
+  select: {
+    marginRight: "10px",
+    height: "50px",
+    width: "20%",
+    fontSize: "18px"
+  },
+  button: {
+    padding: "10px",
+    width: "18%",
+    background: "#0290ff",
+    color: "#fff",
+    padding: "16px",
+    fontSize: "18px"
+  }
+}
+
+
+const bookCategories = [
+  "Action",
+  "Biography",
+  "History",
+  "Horror",
+  "Kids",
+  "Learning",
+  "Sci-Fi"
+];
+
+const generateID = () => {
+  return Math.floor((Math.random() * 1000) + 1);
+}
+
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: 'Action'
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const name = event.target.name;
+
+    this.setState({
+      [name]: event.target.value
+    })
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    this.props.createBook({
+      id: generateID(),
+      title: this.state.title,
+      category: this.state.category
+    });
+
+    this.setState({
+      title: '',
+      category: 'Action'
+    });
+
+    event.target.reset();
```

Comment:
This is unnecessary. The `this.setState` you did above already resets the fields. Oh by the way this `this.setState` call won't be useful until you include `value` attributes to your `input` and `select` elements below.

Reply:
Reminder about this `event.target.reset()` code